projects/bp-gallery/src/components/views/location-curating/LocationPanel.tsx

```diff
@@ -0,0 +1,70 @@
+import { Add } from '@mui/icons-material';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSimplifiedQueryResponseData } from '../../../graphql/queryUtils';
+import useGenericTagEndpoints from '../../../hooks/generic-endpoints.hook';
+import { FlatTag, TagType } from '../../../types/additionalFlatTypes';
+import LocationBranch from './LocationBranch';
+import LocationPanelHeader from './LocationPanelHeader';
+import { useGetTagStructures } from './tag-structure-helpers';
+import { useCreateNewTag } from './location-management-helpers';
+
+const setUnacceptedSubtagsCount = (tag: FlatTag) => {
+  if (!tag.child_tags?.length) {
+    return tag.accepted ? 0 : 1;
+  }
+  let subtagCount = 0;
+  tag.child_tags.forEach((childTag: FlatTag) => {
+    subtagCount += setUnacceptedSubtagsCount(childTag);
+  });
+  tag.unacceptedSubtags = subtagCount;
+  return subtagCount;
+};
+
+const LocationPanel = () => {
+  const { t } = useTranslation();
+
+  const { allTagsQuery } = useGenericTagEndpoints(TagType.LOCATION);
+
+  const { data, refetch } = allTagsQuery();
+  const flattened = useSimplifiedQueryResponseData(data);
+  const flattenedTags: FlatTag[] | undefined = flattened ? Object.values(flattened)[0] : undefined;
+
+  const { createNewTag } = useCreateNewTag(refetch);
+
+  const tagTreeResult = useGetTagStructures(flattenedTags);
+
+  const tagTree = useMemo(() => {
+    if (!tagTreeResult.tagTree) return;
+
+    tagTreeResult.tagTree.forEach(tag => {
+      tag.name, setUnacceptedSubtagsCount(tag);
+    });
+
+    return tagTreeResult.tagTree;
+  }, [tagTreeResult.tagTree]);
+
+  return (
+    <div>
+      <div className='location-panel-header'>
+        <LocationPanelHeader />
+      </div>
+      <div className='location-panel-content'>
+        {tagTree?.map(tag => (
+          <LocationBranch key={tag.id} locationTag={tag} refetch={refetch} />
+        ))}
+        <div
+          className='add-tag-container'
+          onClick={() => {
+            createNewTag(tagTree);
+          }}
+        >
+          <Add className='add-tag-icon' />
+          <div className='add-tag-text'>{t(`tag-panel.add-location`)}</div>
+        </div>
```

Comment:
This could be extracted into a component, since it's used in `LocationBranch`, too.